source/includes/inkling2-reference/_functions.html.md

```diff
@@ -0,0 +1,233 @@
+  '}']* [else '{'
+    [functionStatement]*
+  '}']
+```   
+
+#### Return Statements
+A return statement terminates execution of the function and returns the specified value to its caller. The type of the expression must match the declared return type of the function.
+
+```inkling2--syntax
+returnStatement :==
+  return <expression>
+```
+
+### Expressions
+Inkling supports basic expressions and a variety of unary and binary operators.
+
+#### Built-in Operators
+| Operation        | Syntax        | Description |
+| -                | -             | -           |
+| Function call    | X()           | Invokes a function with zero or more arguments |
+| Array index      | X[Y]          | Indexes into an array |
+| Member access    | X.Y           | References a field within a structure |
+| Neg/pos          | -, +          | Negate numeric value |
+| Logical not      | not           | Logical not (produces 0 or 1) |
+| Exponent         | **            | X raised to the Y power |
+| Mult/div/mod     | *, /, %       | X times Y, X divided by Y, X mod Y |
+| Add/subtract     | +, -          | X plus Y, X minus Y |
+| Relational       | <, <=, >, >=  | Less, less-or-equal, greater, greater-or-equal |
+| Equality         | ==, !=        | Equal, not equal |
+| Logical and      | and           | Left expression if truthy, else right expression |
+| Logical or       | or            | Left expression if falsy, else right expression |
+
+
+As in algebra, parentheses can be used to make order of evaluation explicit. If parentheses are omitted, the table above defines the order of operations.
+
+All mathematical, logical and relational operators work only with numerical operands. Operands of other types (strings, arrays, structures, etc.) are flagged as errors by the compiler.
+
+
+### Function Calls
+A function call temporarily suspends execution of the current function and resumes execution temporarily in the context of a new function. The caller may specify zero or more arguments to be passed to the function. The types of these arguments must be compatible with the declared types of the target function’s parameters.
+
+Inkling does not support direct or indirect recursion. Recursive calls are flagged as errors by the compiler.
+
+```inkling2--syntax
+<subroutine>'('[<expression> ',' ] ')'
+```
+
+#### Literals
+Expression may contain literal values. The following code shows examples of literal values of various types.
+
+```inkling2--code
+# Numeric literal
+var Num = 3.4
+
+# String literal
+var Str = "Hello"
+
+# Array literal
+var Array = [3, 4, 5]
+
+# Struct literal
+var Struct = { X: 3, Y: 4, Z: 4 }
+```   
+
+
+### Math Namespace
+Inkling supports a variety of built-in mathematical operators. It also provides a simple math library to access various useful mathematical constants and functions. These are available through the “Math” namespace, which can be accessed by adding the statement “using Math” to the Inkling program.
```

Comment:
I forget -- does line order matter in Inkling, beyond the first line? e.g. can I put `using Math` all the way at the bottom of the program? The Inkling reference doesn't say.

Reply:
Order doesn't matter for anything except for statements within a function, where the order is obviously important.